### ISSUE TITLE
Cannot remove metadata entries with typed / builder update params

### ISSUE THREAD

**Issue by totof3110** [stripe/stripe-java#893]
Note: I've noticed that with issuing authorizations and transactions but this might be happening with other objects as well.

To try and remove a metadata entry (currently set to `"foo": "bar"`) I try the following:
```java
authorization.update(AuthorizationUpdateParams.builder()
        .putMetadata("foo", null)
        .build());
```
However this does not seem to make any change to the object (the metadata entry is still there with the old value).

I tried calling `.putMetadata("foo", "")` but the SDK rejects empty strings.

The workaround I found was to call:
```java
authorization.update(AuthorizationUpdateParams.builder()
        .putExtraParam("metadata[foo]", EmptyParam.EMPTY)
        .build());
```
which worked.

Should it be possible to remove metadata entries with `putMetadata`?

**Comment by ob-stripe** [comment]
Hi @totof3110. Which version of the library are you using? There was a bug where `null` values in metadata maps were ignored (#788), but it was fixed in v10.9.1.

**Comment by totof3110** [comment]
Hi @ob-stripe . We're on 14.4.1. :-/

**Comment by ob-stripe** [comment]
@totof3110 I was unable to reproduce the issue. Here's the code I tried:

```java
Customer cus = Customer.retrieve("cus_123");

CustomerUpdateParams params = CustomerUpdateParams.builder()
    .putMetadata("foo", null)
    .build();
Customer updatedCus = cus.update(params);

System.out.println(updatedCus.getLastResponse().requestId());
System.out.println(updatedCus.getMetadata().toString());
```

I was able to verify that the metadata key `foo` was properly unset.

I did not try with an authorization object, but metadata should be handled the same way for all resources.

Can you check your dashboard logs to find the issuing authorization update request and see if `"metadata": {"foo": ""}` was sent in the request?

**Comment by totof3110** [comment]
I tried the following:
```java
Transaction transaction = Transaction.retrieve(transactionId, requestOptions);
transaction = transaction.update(TransactionUpdateParams.builder()
		.putMetadata("foo", null)
		.build(), requestOptions);
System.out.println(transaction.getMetadata().get("foo"));
```
which printed out "bar" ("foo" metadata was not removed).

I went to the Stripe logs and the request that was received did not have a body ("No POST body").

**Comment by totof3110** [comment]
The source code for `TransactionUpdateParams.Builder` looks a little different from `ChargeUpdateParams.Builder` so this might be an issue specific to Stripe Issuing authorizations and transactions.

**Comment by ob-stripe** [comment]
Just a quick note that I was able to reproduce the issue. I have a fair idea of what causes this bug, but it's going to be tricky to fix. We'll post an update here when we've made some progress. Thanks for your patience!

**Comment by ob-stripe** [comment]
This should be fixed in the latest version, v15.3.1.
